nixos/modules/tasks/network-interfaces.nix

```
@@ -541,6 +544,25 @@ in
       '';
     };
 
+    networking.machineId = mkOption {
+      default = null;
+      example = "b23d9f80c4494c00b632f2f6448aaa72";
+      type = types.nullOr types.str;
```

```suggestion
      type = types.nullOr (types.strMatching "[[:xdigit:]]{32}");
```

This allows only 32-character hex strings, i.e. correctly formatted16-byte machine-ids. Maybe nicer than allowing any string, to avoid mistakes.